src/gui/entry/EntryView.cpp

```diff
@@ -40,17 +64,117 @@ EntryView::EntryView(QWidget* parent)
     setDragEnabled(true);
     setSortingEnabled(true);
     setSelectionMode(QAbstractItemView::ExtendedSelection);
-    header()->setDefaultSectionSize(150);
 
     // QAbstractItemView::startDrag() uses this property as the default drag action
     setDefaultDropAction(Qt::MoveAction);
 
     connect(this, SIGNAL(doubleClicked(QModelIndex)), SLOT(emitEntryActivated(QModelIndex)));
     connect(selectionModel(), SIGNAL(selectionChanged(QItemSelection,QItemSelection)), SIGNAL(entrySelectionChanged()));
-    connect(m_model, SIGNAL(switchedToEntryListMode()), SLOT(switchToEntryListMode()));
-    connect(m_model, SIGNAL(switchedToGroupMode()), SLOT(switchToGroupMode()));
+    /**
+     * @author Fonic <https://github.com/fonic>
+     * Connect signals to get notified about list/search mode switches (NOTE:
+     * previously named 'switch[ed]ToGroupMode'/'switch[ed]ToEntryListMode')
+     */
+    connect(m_model, SIGNAL(switchedToListMode()), SLOT(switchToListMode()));
+    connect(m_model, SIGNAL(switchedToSearchMode()), SLOT(switchToSearchMode()));
+    /**
+     * @author Fonic <https://github.com/fonic>
+     * Connect signals to notify about changes of view state when state of
+     * 'Hide Usernames'/'Hide Passwords' settings changes in model
+     */
+    connect(m_model, SIGNAL(hideUsernamesChanged()), SIGNAL(viewStateChanged()));
+    connect(m_model, SIGNAL(hidePasswordsChanged()), SIGNAL(viewStateChanged()));
 
     connect(this, SIGNAL(clicked(QModelIndex)), SLOT(emitEntryPressed(QModelIndex)));
+
+    /**
+     * @author Fonic <https://github.com/fonic>
+     * Create header context menu:
+     * - Actions to toggle state of 'Hide Usernames'/'Hide Passwords' settings
+     * - Actions to toggle column visibility, with each action carrying 'its'
+     *   column index as data
+     * - Actions to resize columns
+     * - Action to reset view to defaults
+     */
+    m_headerMenu = new QMenu(this);
+    m_headerMenu->setTitle(tr("Customize View"));
+    m_headerMenu->addSection(tr("Customize View"));
+
+    m_hideUsernamesAction = m_headerMenu->addAction(tr("Hide Usernames"), m_model, SLOT(toggleHideUsernames(bool)));
+    m_hideUsernamesAction->setCheckable(true);
+    m_hidePasswordsAction = m_headerMenu->addAction(tr("Hide Passwords"), m_model, SLOT(toggleHidePasswords(bool)));
+    m_hidePasswordsAction->setCheckable(true);
+    m_headerMenu->addSeparator();
+
+    m_columnActions = new QActionGroup(this);
+    m_columnActions->setExclusive(false);
+    for (int colidx = 1; colidx < header()->count(); colidx++) {
```

Comment:
++colidx, maybe also rename to colIndex.

Reply:
Then lets go all the way and name it columnIndex.

Reply:
Fine by me.

Reply:
Done